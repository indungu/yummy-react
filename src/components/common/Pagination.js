import React from 'react';

export default (props) => {
  const pages = [];
  for (let i = 1; i <= props.pageCount; i++) {
    pages.push(i);
  }
  return (
    <nav aria-label="Page navigation">
      <ul className="pagination justify-content-center">
        { pages.map((page, id) => {
          return (
            <li key={id} className="page-item">
              <button
                className="page-link"
                id={page}
                onClick={props.showPageItems}
              >{page}
              </button>
            </li>
          );
        })}
      </ul>
    </nav>
  );
};
