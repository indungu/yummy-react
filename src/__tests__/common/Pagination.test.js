import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Pagination from '../../components/common/Pagination';

describe('Pagination', () => {
  // eslint-disable-next-line
  const wrapper = shallow(<Pagination pageCount={5}/>);
  it('Renders properly when user is logged in', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
