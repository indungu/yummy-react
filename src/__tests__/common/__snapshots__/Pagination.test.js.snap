// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Renders properly when user is logged in 1`] = `
<nav
  aria-label="Page navigation"
>
  <ul
    className="pagination justify-content-center"
  >
    <li
      className="page-item"
      key="0"
    >
      <button
        className="page-link"
        id={1}
      >
        1
      </button>
    </li>
    <li
      className="page-item"
      key="1"
    >
      <button
        className="page-link"
        id={2}
      >
        2
      </button>
    </li>
    <li
      className="page-item"
      key="2"
    >
      <button
        className="page-link"
        id={3}
      >
        3
      </button>
    </li>
    <li
      className="page-item"
      key="3"
    >
      <button
        className="page-link"
        id={4}
      >
        4
      </button>
    </li>
    <li
      className="page-item"
      key="4"
    >
      <button
        className="page-link"
        id={5}
      >
        5
      </button>
    </li>
  </ul>
</nav>
`;
